storage/segment_query.go

```diff
@@ -349,7 +352,7 @@ func collectGroupedResults(
 	q query.ParsedGroupedQuery,
 	res *query.GroupedResults,
 ) error {
-	filteredGroupByCalcIterator, err := createFilteredGroupByCalcIterator(
+	filteredDocIDIter, groupByIter, calcIter, err := createFilteredGroupByCalcIterator(
```

Comment:
What's the rationale for splitting these up?

Reply:
See the `collectGroupBy...` functions, the `filteredDocIDIter` does not have APIs to expose the underlying values but rather performs the function of moving the `groupByIter` and the `calcIter` in lock step, whereas the individual `groupByIter` and `calcIter` provide APIs to underlying field values.